sport-moment-nft/contracts/EnglishPremierLeague.cdc

```diff
@@ -0,0 +1,843 @@
+        }
+    }
+
+    /// Tag getters
+    ///
+    pub fun getTag(id: UInt64): EnglishPremierLeague.Tag {
+        return EnglishPremierLeague.tagByID[id]!
+    }
+
+    /// A top level Play with a unique ID
+    ///
+    pub struct Play {
+        pub let id: UInt64
+        pub let metadata: {String: String}
+        pub var tagIds: [UInt64]
+
+        init (id: UInt64, metadata: {String: String}, tagIds: [UInt64]) {
+            if let play = EnglishPremierLeague.playByID[id] {
+                self.id = play.id
+                self.metadata = play.metadata
+                self.tagIds = play.tagIds
+            } else {
+                self.id = id
+                self.metadata = metadata
+                self.tagIds = tagIds
+            }
+        }
+    }
+
+    /// Play getters
+    ///
+    pub fun getPlay(id: UInt64): EnglishPremierLeague.Play {
+        return EnglishPremierLeague.playByID[id]!
+    }
+
+    /// A public struct to access Edition data
+    ///
+    pub struct Edition {
+        pub let id: UInt64
+        pub let seriesID: UInt64
+        pub let setID: UInt64
+        pub let playID: UInt64
+        pub var maxMintSize: UInt64?
+        pub let tier: String
+        pub var numMinted: UInt64
+
+       /// member function to check if max edition size has been reached
+        pub fun maxEditionMintSizeReached(): Bool {
+            return self.numMinted == self.maxMintSize
+        }
+
+        init (id: UInt64, seriesID: UInt64, setID: UInt64, playID: UInt64, maxMintSize: UInt64?, tier: String) {
+            if let edition = EnglishPremierLeague.editionByID[id] {
+                self.id = edition.id
+                self.seriesID = edition.seriesID
+                self.setID = edition.setID
+                self.playID = edition.playID
+                self.maxMintSize = edition.maxMintSize
+                self.tier = edition.tier
+                self.numMinted = edition.numMinted
+            } else {
+                self.id = id
+                self.seriesID = seriesID
+                self.setID = setID
+                self.playID = playID
+                self.maxMintSize = maxMintSize
+
+                // If an edition size is not set, it has unlimited minting potential
+                if maxMintSize == 0 {
+                    self.maxMintSize = nil
+                } else {
+                    self.maxMintSize = maxMintSize
+                }
+
+                self.tier = tier
+                self.numMinted = 0 as UInt64
+            }
+        }
+
+        access(contract) fun close() {
+            pre {
+                self.numMinted != self.maxMintSize: "max number of minted moments has already been reached"
+            }
+
+            self.maxMintSize = self.numMinted
+        }
+
+        /// Mint a Moment NFT in this edition
+        /// Note that this will panic if the max mint size has already been reached.
+        ///
+        pub fun mint(): @EnglishPremierLeague.NFT {
```

Comment:
I am confused why this function lives under a struct that can be called by anyone.

Reply:
Yes. Should have been access(contract) but I agree that it really does not make sense in the Edition struct. Have moved it entirely into admin. Thank you.